### PR TITLE
modules: remove tinycbor from ext/ and use west module feature

### PR DESCRIPTION
**Pull request by nashif**
Use tinycbor from the tinycbor repo directly and drop the library from
ext/.

### CODE REVIEW

File: west.yml [hunk 1]
```diff
@@ -35,6 +35,8 @@ manifest:
   projects:
     - name: net-tools
       revision: 30b7efa827b04d2e47840716b0372737fe7d6c92
+    - name: modules/lib/tinycbor
```
Comment (carlescufi) [inline west.yml]: name: tinycbor
path: modules/lib/tinycbor

**Review by carlescufi — CHANGES_REQUESTED** [review]

**Review by mbolivar — APPROVED** [review]
Manifest part looks good now, @carlescufi please revisit

**Comment by mbolivar** [comment]
@nashif just confirming you really meant `modules/lib/tinycbor` and not `modules/lib/encoding/tinycbor`

**Comment by nashif** [comment]
> @nashif just confirming you really meant `modules/lib/tinycbor` and not `modules/lib/encoding/tinycbor`

well, my preference is to limit the depth to 3 levels, should be easier to deal with and navigate. what do you think?

**Comment by mbolivar** [comment]
> well, my preference is to limit the depth to 3 levels, should be easier to deal with and navigate. what do you think?

All good, just confirming. Even `modules/tinycbor` would be fine with me.

**Review by carlescufi — APPROVED** [review]

**Review by tejlmand — APPROVED** [review]
Looking good

**Review by MaureenHelm — APPROVED** [review]

**Review by galak — APPROVED** [review]

**Review by galak — COMMENTED** [review]
Not sure what the exact "west" commands one needs to use now, but can we point at docs for that or something in the commit message.

Also, I'd ask we send an email to the dev list about this change of work flow.

**Review by aescolar — APPROVED** [review]
This change in itself is logical and in line with the agreed direction. So I have no complaints.
( The comment you (@nashif) did during the TSC, made me understand this PR was affecting sanitycheck, which confused me quite a bit )

**Comment by nashif** [comment]
> The comment you (@nashif) did during the TSC, made me understand this PR was affecting sanitycheck, which confused me quite a bit

I mentioned sanitycheck because this is probably the one place you will see the problem with building anything that depends on tinycbor if you are not setup correctly. We have just 1-2 samples/tests that use that and unless you are working on those directly, chances are you will never hit build problem if you are not using west.

**Comment by nashif** [comment]
> Not sure what the exact "west" commands one needs to use now, but can we point at docs for that or something in the commit message.

No commands needs, this should work transparently if you already followed the documentation and setup west.

**Comment by nashif** [comment]
> No commands needed, this should work transparently if you already followed the documentation and setup west.

actually, one command needed

```west update``` to get tinycbor pulled into the zephyr multi-repo tree

**Comment by lgl88911** [comment]
@nashif I using zephyr without west([link](https://docs.zephyrproject.org/latest/guides/west/without-west.html#no-west)), In the case the pr will cause can't find tinycbor. It's means : I must switch to west build the zephyr? Thanks！

**Comment by aescolar** [comment]
@lgl88911 Yes.
Unless you have special needs and know very well what you are doing, you should use west.

**Comment by lgl88911** [comment]
@aescolar Thanks！the west for zepyhr look like repo for android. I'll try it

**Pull request merged**